### ISSUE TITLE
Not working with doctrine/dbal 3.3

### ISSUE THREAD

**Issue by N-M** [itsgoingd/clockwork#583]
After the upgrade from doctrine/dbal 2 to 3.3 Clockwork stopped working.  Removal of code below shows performance tab only.

```
 /** @var \Clockwork\Clockwork $clock */
$clock = $container->get('clockwork');
$clock->addDataSource(new DoctrineDataSource($entityManager));
```

![image](https://user-images.githubusercontent.com/781417/177134364-52f990de-2030-4097-a101-035a93c72b80.png)

**Comment by itsgoingd** [comment]
Hey, can you please provide more information about your app? From the code example it doesn't seem like a standard Laravel app.

**Comment by N-M** [comment]
It is the Slim app with Clockwork middleware.

I have made a test repo. This is a working example with dbal2 https://github.com/N-M/Slim-Skeleton
![image](https://user-images.githubusercontent.com/781417/177647994-21b87d9d-9a5d-4d4a-b68a-2a22d1181bfb.png)

An example with dbal3 that is not working can be found here https://github.com/N-M/Slim-Skeleton/tree/dbal3
![image](https://user-images.githubusercontent.com/781417/177648195-360f5716-06ca-4ab1-afe4-16ee9da6acc2.png)


Please let me know if you need more details

**Comment by itsgoingd** [comment]
Thanks, this is very useful, will return to this tomorrow.

**Comment by itsgoingd** [comment]
Hey, I've finally got around to check this out. This should now be fixed in master and will be released soon, sorry for the long wait.

This was actually caused by resolving of the connection name for executed queries. In DBAL 2 this was implemented by looking at the configuration, this implementation changed in DBAL 3 to execute an actual query, thus leading to infinite recursion.

I've fixed this by resolving the connection name only once, early in the process, instead on every collected query.
